### ISSUE TITLE
ios: Local images are not rendering in 0.62.2

### ISSUE THREAD

**Issue by rahamin1** [facebook/react-native#29237]
Please provide all the information requested. Issues that do not follow this format are likely to stall.

## Description
I am rendering local images, using the 'Image' component, on android, with path like  file:///data/user/0/com.xxxx/files/productdata_da13681addffaed7622ece9b2bb56e22bc65f6b9.jpg.

On iOS, however, these images are not rendering.

I tried the suggestion in https://stackoverflow.com/a/54627588/5532513, didn't help.

## React Native version:
System:
    OS: macOS 10.15.5
    CPU: (8) x64 Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz
    Memory: 1.49 GB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.16.2 - /usr/local/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.14.4 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.9.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 13.5, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
    Android SDK: Not Found
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6514223
    Xcode: 11.5/11E608c - /usr/bin/xcodebuild
  Languages:
    Java: Not Found
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.11.0 => 16.11.0 
    react-native: 0.62.2 => 0.62.2 

## Steps To Reproduce
Just using:

```
<Image
        style={{ width: 50, height: 50 }}
        source={{ uri: mobileImageFilePath }}
      />
```

## Expected Results
Images should be displayed.

**Comment by warrenalphonso** [comment]
Yup I'm getting the same problem. I made a new project with `react-native init`, added a local file, and then used a `<Image />` component. It works as expected on Android, but not on iOS. Doesn't work with URIs either.

**Comment by TrustTheBoy** [comment]
OS: macOS 11
Platforms: iOS 14.0
Xcode: 12.0
react-native: 0.62.2 => 0.62.2
<Image /> iOS14.0 cannot display pictures, iOS13.5 displays normally

**Comment by esensei** [comment]
Same issue, ios 14.

**Comment by artmerlop** [comment]
Same issue here!

**Comment by rahamin1** [comment]
Are you also having a problem only with local files that your application is storing? 
(I would like to emphasize that I don't have any problems with assets)

I am using RNFetchBlob to store these files. Do you? The problem may be different behavior of RNFetchBlob on iOS compared to android.

**Comment by simonarcher** [comment]
I have this problem with local assets in my RN app project, as well as rendering remote assets with a url.

**Comment by rahamin1** [comment]
As I wrote above, in my case local assets work fine, and I am not using url to display images.

**Comment by rahamin1** [comment]
I am closing this issue. I need to do some more tests.
I suggest that you open new issue(s) for your problems. It seems that the at least some of these problems are different than mine.

**Comment by esensei** [comment]
> I have this problem with local assets in my RN app project, as well as rendering remote assets with a url.

Hello, you solved the problem with local image paths? I don’t understand why the author closed the ticket

**Comment by rahamin1** [comment]
I didn't solve the problem yet, but I realized that I need to do more tests before posting an issue.

At least some of the comments reported other issues - like displaying assets and rendering from a URL, so they should open other issues. You can also open an issue, explaining your problem.

**Comment by tachtevrenidis** [comment]
I have the same issue

**Comment by abiespinal04** [comment]
I have same issue with iOS 14. It doesn't show local images in my react native app.

**Comment by lh9403** [comment]
so, how to solve it in iOS 14?

**Comment by MoOx** [comment]
This has been fixed in 0.63.2 See https://github.com/react-native-community/releases/blob/master/CHANGELOG.md#v0632

**Comment by rahamin1** [comment]
I am reopening this, until we tests it in 0.63.2 and see that the problem has been solved.

**Comment by lukeramsden** [comment]
Could we get a hotfix for 0.62?

**Comment by cristianoccazinsp** [comment]
Looking for a 0.62 emergency fix for this as well.  Way to go Apple, way to go...

**Comment by anantanandgupta** [comment]
i can't update the react-native to 63 yet... need a fix on 62

**Comment by lukeramsden** [comment]
For anyone that cant't upgrade to 62 like me and need to fix this, get yourself set up with https://github.com/ds300/patch-package and then put this in your patches folder:

`react-native+0.62.2.patch`:

```diff
diff --git a/node_modules/react-native/Libraries/Image/RCTUIImageViewAnimated.m b/node_modules/react-native/Libraries/Image/RCTUIImageViewAnimated.m
index 21f1a06..0ff66f3 100644
--- a/node_modules/react-native/Libraries/Image/RCTUIImageViewAnimated.m
+++ b/node_modules/react-native/Libraries/Image/RCTUIImageViewAnimated.m
@@ -275,6 +275,8 @@ - (void)displayLayer:(CALayer *)layer
   if (_currentFrame) {
     layer.contentsScale = self.animatedImageScale;
     layer.contents = (__bridge id)_currentFrame.CGImage;
+  } else {
+    [super displayLayer:layer];
   }
 }
```

**Comment by rjtubera** [comment]
Hotfix for RN 0.61.2 please. I cannot make it work, even my bottom navigator doesn't have images

**Comment by Andrea-Arguello** [comment]
> For anyone that cant't upgrade to 62 like me and need to fix this, get yourself set up with https://github.com/ds300/patch-package and then put this in your patches folder:
> 
> `react-native+0.62.2.patch`:
> 
> ```diff
> diff --git a/node_modules/react-native/Libraries/Image/RCTUIImageViewAnimated.m b/node_modules/react-native/Libraries/Image/RCTUIImageViewAnimated.m
> index 21f1a06..0ff66f3 100644
> --- a/node_modules/react-native/Libraries/Image/RCTUIImageViewAnimated.m
> +++ b/node_modules/react-native/Libraries/Image/RCTUIImageViewAnimated.m
> @@ -275,6 +275,8 @@ - (void)displayLayer:(CALayer *)layer
>    if (_currentFrame) {
>      layer.contentsScale = self.animatedImageScale;
>      layer.contents = (__bridge id)_currentFrame.CGImage;
> +  } else {
> +    [super displayLayer:layer];
>    }
>  }
> ```

This worked for me for Debug on RN 0.61.5, but it won't work for Release! Any idea? I've even downgraded from Xcode 12 to 11.1 to no avail.

**Comment by lukeramsden** [comment]
> This worked for me for Debug on RN 0.61.5, but it won't work for Release! Any idea why? I've even downgraded from Xcode 12 to 11.1, hoping that would solve the issue, but it didn't. This is the output of `xcodebuild -showsdks`

If you've used my patch verbatim then it's probably not working because you're on a different version and patch package won't like that.
Regardless, as far as I am aware the issue is only on iOS 14, so it works in Debug mode because you're running on an iOS simulator that is running version 13. If you can't upgrade to a fixed version of RN< then go to `node_modules/react-native/Libraries/Image/RCTUIImageViewAnimated.m` and add that extra else statement, then run `patch-package`. Read the docs for `patch-package` to better understand what I mean.

**Comment by Andrea-Arguello** [comment]
> > This worked for me for Debug on RN 0.61.5, but it won't work for Release! Any idea why? I've even downgraded from Xcode 12 to 11.1, hoping that would solve the issue, but it didn't. This is the output of `xcodebuild -showsdks`
> 
> If you've used my patch verbatim then it's probably not working because you're on a different version and patch package won't like that.
> Regardless, as far as I am aware the issue is only on iOS 14, so it works in Debug mode because you're running on an iOS simulator that is running version 13. If you can't upgrade to a fixed version of RN then go to `node_modules/react-native/Libraries/Image/RCTUIImageViewAnimated.m` and add that extra else statement, then run `patch-package react-native`. Read the docs for `patch-package` to better understand what I mean.

Yes, I've done the changes manually and applied ```patch-package react-native``` to keep them, to no avail.

**Comment by lukeramsden** [comment]
It's likely to be a caching issue somewhere in the build process I would wager. Not much else I can do to help, my patch was just the following commit: https://github.com/facebook/react-native/commit/b6ded7261544c703e82e8dbfa442dad4b201d428 I don't have any real deeper understanding of this issue.

**Comment by Andrea-Arguello** [comment]
It is a bit obscure, as we've tried doing it in another computer with Xcode 11.1, and the issue still persists. On Android it does build correctly, I'll try with a blank slate project and see if it could be any of the Cocoapods libraries I use that's causing the issue.

**Comment by olivergeorge** [comment]
The 0.63.2 release includes a two line fix.  Unfortunately that commit wasn't included in the 0.62.3 release.

Fix image cannot show in iOS 14 (123423c2a9)

Manually applying this change fixed image loading for me.  (RN62.3 + xcode 12.5)
